### ISSUE TITLE
Help getting GHE to work?

### ISSUE THREAD

**Issue by gesa** [gharlan/alfred-github-workflow#78]
This is the absolute worst bug report of all time but… I can't get ghe to work. I have my url set, i've added my access token via `ghe > login 6f8[etcetcetc]`, I've gotten the notification stating "Successfully logged in", but any time i type `ghe`, these are my only options: 
![20170124191853](https://cloud.githubusercontent.com/assets/782056/22276706/53c9dc12-e26a-11e6-9f47-3607eda6dc52.png)

I tired starting over (enterprise reset, url, login) but to the same result. This has been a persistent problem for some time now, so reboot isn't the issue either. Additionally, I've tried it with several different personal tokens, just in case i screwed that up. Workflow is up-to-date, OS is 10.12.3.

(ghe URL and personal token have been redacted. both debug logs were fresh starts—i ran `ghe > enterprise reset` before beginning logging.)

<details>
<summary>Debug log, set to "interesting information" level.</summary>

```
[2017-01-24 19:25:12][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:12][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:14][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:14][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:14][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:14][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:14][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][ERROR: input.scriptfilter] Code 15: using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:15][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:16][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:16][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:16][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:16][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:18][ERROR: input.scriptfilter] Code 15: using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:18][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:19][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:20][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:25:21][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:26:50][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
```
</details>

<details>
<summary>Debug log, set to "all information" level.</summary>

```
[2017-01-24 19:32:44][input.scriptfilter] <?xml version="1.0"?>
<items/>
[2017-01-24 19:32:45][STDERR: input.scriptfilter] loading content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:45][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="c7f4909480924b60440d2144c4540984" arg="e &gt; enterprise-url " autocomplete=" &gt; url " valid="no"><icon>icon.png</icon><title>&gt; url &lt;URL&gt;</title><subtitle>Set the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:32:46][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:46][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="0cf7458e27a77dbbd291bce8f6abd6cb" arg="e &gt; enterprise-url " autocomplete=" &gt; url " valid="no"><icon>icon.png</icon><title>&gt; url &lt;URL&gt;</title><subtitle>Set the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:32:47][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:47][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="df98a340e25c624b4761ace38c8f5c3e" arg="e &gt; enterprise-url github.[REDACTED].com" autocomplete=" &gt; url github.[REDACTED].com"><icon>icon.png</icon><title>&gt; url github.[REDACTED].com</title><subtitle>Set the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:32:49][input.scriptfilter] Processing output of 'action.script' with arg 'e > enterprise-url github.[REDACTED].com'
[2017-01-24 19:32:50][action.script] Processing output of 'output.notification' with arg ''
[2017-01-24 19:32:50][input.scriptfilter] <?xml version="1.0"?>
<items/>
[2017-01-24 19:32:50][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:50][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="a3409aa6ba3acd3547577382c931e30e" arg="e &gt; login " autocomplete=" &gt; login " valid="no"><icon>icon.png</icon><title>&gt; login &lt;access_token&gt;</title><subtitle>Save access token</subtitle></item><item uid="6820c6b4fd6f3318c08d20b95e5779bb" arg="github.[REDACTED].com/settings/applications" autocomplete=" &gt; generate token"><icon>icon.png</icon><title>&gt; generate token</title><subtitle>Generate a new access token</subtitle></item><item uid="25bf98b3344410c9a8fe78a7aa3ffc21" arg="e &gt; enterprise-reset" autocomplete=" &gt; enterprise reset"><icon>icon.png</icon><title>&gt; enterprise reset</title><subtitle>Reset the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:32:52][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:52][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="5c8bbbd34ddb0bd9aab441233ae47e92" arg="e &gt; login " autocomplete=" &gt; login " valid="no"><icon>icon.png</icon><title>&gt; login &lt;access_token&gt;</title><subtitle>Save access token</subtitle></item><item uid="5713c7bd6882ce557c68c9512d5d5d17" arg="github.[REDACTED].com/settings/applications" autocomplete=" &gt; generate token"><icon>icon.png</icon><title>&gt; generate token</title><subtitle>Generate a new access token</subtitle></item><item uid="a6f4952f7240a333e520e563e5012c4a" arg="e &gt; enterprise-reset" autocomplete=" &gt; enterprise reset"><icon>icon.png</icon><title>&gt; enterprise reset</title><subtitle>Reset the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:32:56][input.scriptfilter] <?xml version="1.0"?>
<items/>
[2017-01-24 19:32:56][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:32:56][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="696aacb99a44c03fc8daf84e312f9454" arg="e &gt; login " autocomplete=" &gt; login " valid="no"><icon>icon.png</icon><title>&gt; login &lt;access_token&gt;</title><subtitle>Save access token</subtitle></item><item uid="5bd8626fb0908ca7edc43e78034685f9" arg="github.[REDACTED].com/settings/applications" autocomplete=" &gt; generate token"><icon>icon.png</icon><title>&gt; generate token</title><subtitle>Generate a new access token</subtitle></item><item uid="b56c449d07d52eb30fae367cddcbd2ec" arg="e &gt; enterprise-reset" autocomplete=" &gt; enterprise reset"><icon>icon.png</icon><title>&gt; enterprise reset</title><subtitle>Reset the GitHub Enterprise URL</subtitle></item></items>
[2017-01-24 19:33:00][STDERR: input.scriptfilter] using cached content for https://api.github.com/repos/gharlan/alfred-github-workflow/releases/latest
[2017-01-24 19:33:00][input.scriptfilter] <?xml version="1.0"?>
<items><item uid="ac71dc631274a0e59a8de20b14b9ac44" arg="e &gt; login [REDACTED]" autocomplete=" &gt; login [REDACTED]"><icon>icon.png</icon><title>&gt; login [REDACTED]</title><subtitle>Save access token</subtitle></item></items>
[2017-01-24 19:33:00][input.scriptfilter] <?xml version="1.0"?>
<items/>
[2017-01-24 19:33:01][input.scriptfilter] Processing output of 'action.script' with arg 'e > login [REDACTED]'
[2017-01-24 19:33:01][action.script] Processing output of 'output.notification' with arg 'Successfully logged in'
```
</details>

Do you suppose you could help me out with this? I think i might die without this alfred script. i've been using for, idek, as long as time has existed?

**Comment by gharlan** [comment]
Please try to open this URL in browser:
`<ghe-host>/api/v3/user?access_token=<your-access-token>`

Do you get details about your user?

**Comment by gesa** [comment]
@gharlan I do!

**Comment by gharlan** [comment]
hmm hmm, strange.
Maybe try to delete the data directory of the workflow (`/Users/$USER/Library/Application Support/Alfred 3/Workflow Data/de.gh01.alfred.github`) and try again.

**Comment by gesa** [comment]
I got to the bottom of it. It was VPN related.

My company's instance of GitHub Enterprise *can* be accessed from outside the VPN, but certain features (I know "uploading images" is covered by this for example, but idk what else) are protected by the VPN.

I followed your instruction and tried to log in again. Once again the notification told me login was successful, but I was not logged in. I then tried the same steps on the VPN and it worked. I did a couple of searches in the workflow, they worked great.

i then disconnected from the VPN. the cached searches still showed up and, as a result, were functional in taking me to where i needed to go, but unrelated searches didn't work. the good news though is that at least hitting "enter" would take me to my ghe instance either way.

thanks for your help, and sorry i didn't think of this earlier. i only recently realized that uploading images off-network didn't work.
